pkg/api/types.go

```diff
@@ -153,6 +153,15 @@ const (
 	TerminationMessagePathDefault string = "/dev/termination-log"
 )
 
+// There is no namespace required for these kinds specified here.
+// If a kind is not enumerated here, it is assumed to have a namespace scope.
+var NoneNamespacedKinds = []string{
```

Comment:
Ultimately it's the strategies responsibility - whatever has to be injected
into the strategy to make that determination is usually fine.  For
instance, we validate references for access control on mutation via the
strategy on Openshift image streams (you cannot reference an item you don't
have visibility on).

On Oct 24, 2015, at 11:46 PM, Tim Hockin <notifications@github.com> wrote:

In pkg/api/types.go
<https://github.com/kubernetes/kubernetes/pull/16146#discussion_r42940591>:

> @@ -153,6 +153,15 @@ const (
>  	TerminationMessagePathDefault string = "/dev/termination-log"
>  )
>
> +// There is no namespace required for these kinds specified here.
> +// If a kind is not enumerated here, it is assumed to have a namespace scope.
> +var NoneNamespacedKinds = []string{

Yeah, that doesn't bode well for this PR. To do this sanely (at least as I
can see it) involves doing something like making a validator context which
the validation functions take (as an arg or as a method receiver). That's
no longer a small fix.

@lavalamp <https://github.com/lavalamp> since he had some thoughts about
validation perhaps being 2 passes - one to validate the object with no
external state and another to validate against other objects or the system
state. This sort of falls in the middle..

—
Reply to this email directly or view it on GitHub
<https://github.com/kubernetes/kubernetes/pull/16146/files#r42940591>.